### PR TITLE
feat: add line_width property to object display

### PR DESCRIPTION
**Pull request by tkimura4**
Signed-off-by: tomoya.kimura <tomoya.kimura@tier4.jp>

## Description
Add a line with property to object display
image

## Related links
https://github.com/tier4/autoware_launch/pull/378

## Review procedure
Check that the line width of object-shape changes depending on the value of line witdh property on rviz.

(with https://github.com/tier4/autoware_launch/pull/378 )

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_display_base.hpp [hunk 1]
```diff
@@ -138,12 +139,12 @@ class AUTOWARE_AUTO_PERCEPTION_RVIZ_PLUGIN_PUBLIC ObjectPolygonDisplayBase
   std::optional<Marker::SharedPtr> get_shape_marker_ptr(
     const autoware_auto_perception_msgs::msg::Shape & shape_msg,
     const geometry_msgs::msg::Point & centroid, const geometry_msgs::msg::Quaternion & orientation,
-    const ClassificationContainerT & labels) const
+    const ClassificationContainerT & labels, const double line_width = 0.03) const
```
Comment (shmpwk) [inline common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_display_base.hpp:142]:
```suggestion
    const ClassificationContainerT & labels, const double & line_width) const
```

I'm not sure but it is natual for me.

File: common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_detail.hpp [hunk 2]
```diff
@@ -80,7 +80,7 @@ AUTOWARE_AUTO_PERCEPTION_RVIZ_PLUGIN_PUBLIC visualization_msgs::msg::Marker::Sha
 get_shape_marker_ptr(
   const autoware_auto_perception_msgs::msg::Shape & shape_msg,
   const geometry_msgs::msg::Point & centroid, const geometry_msgs::msg::Quaternion & orientation,
-  const std_msgs::msg::ColorRGBA & color_rgba);
+  const std_msgs::msg::ColorRGBA & color_rgba, const double line_width = 0.03);
```
Comment (shmpwk) [inline common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_detail.hpp:83]:
```suggestion
  const std_msgs::msg::ColorRGBA & color_rgba, const double & line_width);
```

I'm not sure but it is natual for me.

**Review by shmpwk — COMMENTED** [review]
Adding Parameters description is better

https://github.com/autowarefoundation/autoware.universe/blob/36c323b4a6794f8fa78f9898262892ba964bf3e2/common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_detail.hpp#L77-L78



https://github.com/autowarefoundation/autoware.universe/blob/36c323b4a6794f8fa78f9898262892ba964bf3e2/common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_display_base.hpp#L136-L137


```
/// \param line_width Line thickness around the object
```

File: common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_detail.hpp (hunk 2 above)
Comment (shmpwk) [inline common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_detail.hpp:83]: I am confusing why we have to specify default value here. (But actually rviz crush without default value.)

File: common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp [hunk 3]
```diff
@@ -212,7 +212,7 @@ visualization_msgs::msg::Marker::SharedPtr get_label_marker_ptr(
 visualization_msgs::msg::Marker::SharedPtr get_shape_marker_ptr(
   const autoware_auto_perception_msgs::msg::Shape & shape_msg,
   const geometry_msgs::msg::Point & centroid, const geometry_msgs::msg::Quaternion & orientation,
-  const std_msgs::msg::ColorRGBA & color_rgba)
+  const std_msgs::msg::ColorRGBA & color_rgba, const double line_width)
```
Comment (shmpwk) [inline common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp:215]:
```suggestion
  const std_msgs::msg::ColorRGBA & color_rgba, const double & line_width)
```

reference value is better?

File: common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_detail.hpp (hunk 2 above)
Comment (tkimura4) [reply common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_detail.hpp]: We didn't need to specify a value, but 0.03 was originally used as line_width, so I set it as default value.
image

I removed default value at [652a33a](https://github.com/autowarefoundation/autoware.universe/pull/1206/commits/652a33a8bf3040a7963f820fa8fc5868e54504d6) and confirmed that it works well without crushing

Comment (shmpwk) [reply common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_detail.hpp]: I tried again and rviz does not crush.
So [652a33a](https://github.com/autowarefoundation/autoware.universe/pull/1206/commits/652a33a8bf3040a7963f820fa8fc5868e54504d6) looks good for me!

**Review by shmpwk — APPROVED** [review]

**Pull request merged**